pkg/server/auto_upgrade.go

```diff
@@ -52,6 +52,11 @@ func (s *topLevelServer) startAttemptUpgrade(ctx context.Context) error {
 		}
 
 		for r := retry.StartWithCtx(ctx, retryOpts); r.Next(); {
+			if isMeta1, err := s.sqlServer.isMeta1Leaseholder(ctx, s.clock.NowAsClockTimestamp()); err == nil && !isMeta1 {
```

Comment:
downside of this is a node that is never meta1 will never hit `case UpgradeAlreadyCompleted` and exit the loop, so this loop would run forever. OTOH, to get to case UpgradeAlreadyCompleted, we need to read the version, and having all the nodes read the version every x seconds could get us back to contending.

I suppose another option would be an AOST read of the version first -- if it is already upgraded as of -30s, exit. Then check meta1, then go re-read it transactionally at now if we're the meta1 LH.

But not sure if that's worth it, compared to just calling isMeta1LH twice a minute.

Reply:
Doesn't this always add 30s to the startup of every node other than the meta1 leaseholder? 

At a minimum, I would add a big comment about why this is done the way it is, but I'm still not sure I like it very much. 

An AOST or possibly better `INCONSISTENT` read for this operation at startup seems better. In either case, it should be fine if we assume that versions never go backward.

Reply:
>  add 30s to the startup of every node other than the meta1 leaseholder?

How? server-start shouldn't be blocking on any of this unless there's a read where else that hits contention, but if anything this would make that better right?

Reply:
> An AOST or possibly better INCONSISTENT read for this operation at startup seems better. 

I think there are several alternatives we _could_ pursue: we already enumerate all the nodes to check their binary version before we start upgrading, so we could also only then act on the go/no-go outcome of that if we were the maximum or minimum node ID, which would work in tenants as well as the system tenant. 

I'm not actually sure we want to use inconsistent scans or back-dated reads to tell if the version has been bumped, unless you specifically mean in my early exit before checking if we're the special upgrader? I don't think they'd help us with a all-nodes-try-to-upgrade-at-once approach like we have now, and would actually if just used on their own make it worse right? If we only made the scan of the current version more stale, they'd continue to see it as eligible for upgrade and even more nodes would then be competing to do the upgrade and actually (atomically) move the version key, unless I've misunderstood your suggestion?